js-miniapp-bridge/src/common-bridge.ts

```diff
@@ -261,6 +273,22 @@ export class MiniAppBridge {
       );
     });
   }
+
+  /**
+   * Associating fetch function to MiniAppBridge object.
+   *
+   * @param request request informations
+   */
+  fetch(request: FetchRequest) {
+    return new Promise<NativeFetchResponse>((resolve, reject) => {
+      return this.executor.exec(
+        'fetch',
+        request,
+        success => resolve(JSON.parse(success) as NativeFetchResponse),
+        error => reject(JSON.parse(error) as NativeFetchResponse)
```

Comment:
I think `error` should pass a string and not a `Response`. This is because the Promise will only be rejected if there was some issue with sending the request or receiving a response (i.e. the user set invalid params, tried to connect to a URL which isn't in the "Allow List", device is not connected to network, etc.) So in that case, there will be no `Response` object to return to the user.

If the server returns an error response (i.e. 500), then the Promise will be resolved with the error response - this is why the `Response` model has an `ok` property to check if the response is successful.

Reply:
👍  makes sense. Thanks